---
title: "C26111 | Microsoft Docs"
ms.custom: ""
ms.date: "11/04/2016"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "vs-devops-test"
ms.tgt_pltfrm: ""
ms.topic: "article"
f1_keywords: 
  - "C26111"
helpviewer_keywords: 
  - "C26111"
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
caps.latest.revision: 8
author: "corob-msft"
ms.author: "corob"
manager: "ghogen"
caps.handback.revision: 8
---
# C26111
[!INCLUDE[vs2017banner](../code-quality/includes/vs2017banner.md)]

avertissement C26111 : L'appellant n'as pas réussi à garder le vérou \<lock\> avant l'appel de la fonction \<func\>.  
  
 L'annotation `_Requires_lock_not_held_` applique une condition préalable qui le nombre de verrous du verrou spécifié ne peut pas être supérieur à zéro lorsque la fonction est appelée.  C26111 avertissement est émis lorsque la fonction ne libère pas tous les verrous avant qu'elle appelle une autre fonction.  
  
## Exemple  
 L'exemple suivant génère C26111 d'avertissement parce que la condition préalable de `_Requires_lock_not_held_` est violée par l'appel à `DoNotLock` dans la section verrouillée.  
  
```  
  
typedef struct _DATA   
{  
    CRITICAL_SECTION cs;  
    int d;  
} DATA;  
  
_Requires_lock_not_held_(p->cs)   
  
void DoNotLock(DATA* p)  
{  
    EnterCriticalSection(&p->cs);   
    p->d = 0;  
    LeaveCriticalSection(&p->cs);  
}  
  
void LockedFunction(DATA* p)   
{   
    EnterCriticalSection(&p->cs);   
    DoNotLock(p); // Warning C26111  
    LeaveCriticalSection(&p->cs);  
}  
  
```